python/mxnet/gluon/utils.py

```diff
@@ -72,7 +73,7 @@ def split_data(data, num_slice, batch_axis=0, even_split=True):
             "uneven partitioning of data."%(
                 str(data.shape), num_slice, batch_axis, num_slice))
 
-    step = size // num_slice
+    step = math.ceil(1.0 * size / num_slice)
```

Comment:
I am not sure this fixes the problem, for instance when `size=34` and `num_slice=8`, the above calculation will return `step = 5`. How will 34 elements be split 8 ways with a `step = 5`.